### ISSUE TITLE
AddTrack/Add*Transceiver/RemoveTrack concurrency issues during negotiation.

### ISSUE THREAD

**Issue by sgotti** [pion/webrtc#1169]
### Your environment.
  - Version: master eab7996d848f2cc1f16c1ec9aad2eac3ef224f0f
  - Browser: Not related to a browser but reproducible with Chrome at probably any version (tested with 81/82/83/84)

### What did you do?

I noticed this while building an sfu and testing it with chrome (looks like firefox is less susceptible to such issues).

Basically, calling AddTrack/Add*Transceiver/RemoveTrack during a negotiation or, also if the negotiation is ended but the goroutine spawned by `StartTransports`/`StartRenegotiation` is yet running and calling `StartRTPSender`, will create different kind of issues. I'll report two of the possible cases a discovered in my tests:

1. If we're offering and then we call `AddTrack` before receiving an answer/calling `SetRemoteDescription(answer)` the `StartTransports`/`StartRenegotiation` method will start some transceivers not negotiated and so some rtpSenders not provided in the offered sdp. In chrome this will cause an obscure exception when calling `setRemoteDescription` :

```
"Failed to set local answer sdp: Failed to process the bundled m= section"
```

2. If we already offered, received the answered and called SetRemoteDescription(answer)` BUT the `StartTransports`/`StartRenegotiation` spawned goroutines are not finished and we call `AddTrack` that will add a Sender to an existing transport currently without a sender (implemented in bddbfa7e11efd0c0a134ecbd677377a7555c68b1).

Chrome will not report an exception but the stream won't start and in the log we'll find something like: 

```
[3043620:16:0424/144525.744681:ERROR:webrtc_video_engine.cc(3115)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[3043620:16:0424/144525.773075:ERROR:call.cc(1257)] receive_rtp_config_ lookup failed for ssrc 1491190739
[3043620:16:0424/144525.781252:ERROR:rtp_stream_receiver_controller.cc(26)] RtpStreamReceiverController::Receiver::Receiver: Sink could not be added for SSRC=1491190739.
```

Looks like starting a rtpsender before negotiating it is a known issue with chrome: https://bugs.chromium.org/p/webrtc/issues/detail?id=10139
basically starting a rtpsender providing a ssrc not available in the current local sdp will cause issue 1. Issue 2. is more subtle since we don't get any exception but the related track won't be added.

### Possible solution.

**TL;DR**

Only start sender if the both the transceiver and the sender were the one negotiated.

The above issues cannot be avoided with just a "clever" use of the pion/webrtc API by avoid calling AddTrack/Add*Transceiver/RemoveTrack between a negotiation because the `StartTransports`/`StartRenegotiation` goroutines will continue after `SetRemoteDescription` has returned.

**Comment by sgotti** [comment]
Created PR #1170.

**Comment by jeremija** [comment]
@sgotti @Sean-Der is this the same as https://github.com/pion/webrtc/issues/1168 ?

**Comment by sgotti** [comment]
@jeremija Yeah it looks the same, I probably opened it just a bit later than you (I was digging in it for quite a week...).

**Comment by barkhasikka** [comment]
Using pion v2 and still having the same issue. Have extended using pion-sfu to support many to many. OnNegotiationNeeded is fired when a new track is added to subscriber connection. In onNegotiationNeeded we create a new offer and send that to chrome.
